src/main/java/com/minecolonies/coremod/commands/AbstractSingleCommand.java

```diff
@@ -57,4 +69,97 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
+     * @param colonyId the id of the colony.
+     * @return boolean
+     */
+
+    public boolean canPlayerUseCommand(final EntityPlayer player, final Commands theCommand, final int colonyId)
+    {
+        if (isPlayerOpped(player, theCommand.toString()))
+        {
+            return true;
+        }
+
+        final Colony chkColony = ColonyManager.getColony(colonyId);
+        if(chkColony == null)
+        {
+            return false;
+        }
+        return canCommandSenderUseCommand(theCommand)
+                && canRankUseCommand(chkColony, player);
+    }
+
+    /**
+     * Checks if the player has the permission to use the command.
+     * By default officer and owner, overwrite this if other required.
+     * @param colony the colony.
+     * @param player the player.
+     * @return true if so.
+     */
+    public boolean canRankUseCommand(@NotNull final Colony colony, @NotNull final EntityPlayer player)
+    {
+        return colony.getPermissions().getRank(player).equals(Permissions.Rank.OFFICER) || colony.getPermissions().getRank(player).equals(Permissions.Rank.OWNER);
+    }
+
+    /**
+     * Will check the config file to see if players are allowed to use the command that is sent here.
+     *
+     * @param theCommand which command to check if the player can use it
+     * @return boolean
+     */
+    public boolean canCommandSenderUseCommand(Commands theCommand)
+    {
+        switch (theCommand)
+        {
+            case CITIZENINFO:
+                return Configurations.canPlayerUseCitizenInfoCommand;
+            case COLONYTP:
+                return Configurations.canPlayerUseRTPCommand;
+            case KILLCITIZENS:
+                return Configurations.canPlayerUseKillCitizensCommand;
+            case LISTCITIZENS:
+                return Configurations.canPlayerUseListCitizensCommand;
+            case RESPAWNCITIZENS:
+                return Configurations.canPlayerRespawnCitizensCommand;
+            case SHOWCOLONYINFO:
+                return Configurations.canPlayerUseShowColonyInfoCommand;
+            case ADDOFFICER:
+                return Configurations.canPlayerUseAddOfficerCommand;
+            case DELETECOLONY:
+                return Configurations.canPlayerUseDeleteColonyCommand;
+            case REFRESH_COLONY:
+                return Configurations.canPlayerUseRefreshColonyCommand;
+        }
+
+        return false;
+    }
+
+    /**
+     * Will check to see if play is Opped for the given command name.
+     *
+     * @param sender to check the player using the command.
+     * @param cmdName the name of the command to be checked.
+     * @return boolean
+     */
+    @NotNull
+    public boolean isPlayerOpped(@NotNull final ICommandSender sender, String cmdName)
+    {
+        int requiredOpLevel = PERMNUM;
+        if (PERMNUM < 1)
+        {
+            requiredOpLevel = 1;
+        }
+        if (PERMNUM > 4)
+        {
+            requiredOpLevel = 4;
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)